local/network.go

```diff
@@ -206,34 +211,45 @@ func newNetwork(
 	return net, nil
 }
 
-// GenerateDefaultNetwork call generateDefaultNetwork with no mocking
-func GenerateDefaultNetwork(
+// NewDefaultNetwork returns a new network using a pre-defined
+// network configuration.
+// The following addresses are pre-funded:
+// X-Chain Address 1:     X-custom18jma8ppw3nhx5r4ap8clazz0dps7rv5u9xde7p
+// X-Chain Address 1 Key: PrivateKey-ewoqjP7PxY4yr3iLTpLisriqt94hdyDFNgchSxGGztUrTXtNN
+// X-Chain Address 2:     X-custom16045mxr3s2cjycqe2xfluk304xv3ezhkhsvkpr
+// X-Chain Address 2 Key: PrivateKey-2fzYBh3bbWemKxQmMfX6DSuL2BFmDSLQWTvma57xwjQjtf8gFq
+// C-Chain Address:       0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC
+// C-Chain Address Key:   56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027
+// The following nodes are validators:
+// * NodeID-7Xhw2mDxuDS44j42TCB6U5579esbSt3Lg
+// * NodeID-MFrZFVCXPv5iCn6M9K6XduxGTYp891xXZ
+// * NodeID-NFBbbJ4qCmNaCzeW7sxErhvWqvEQMnYcN
+// * NodeID-GWPcbFJZFfZreETSoWjPimr846mXEKCtu
+// * NodeID-P7oB2McjBGgW2NXXWVYjV8JEDFoW9xDE5
+func NewDefaultNetwork(
 	log logging.Logger,
 	binaryPath string,
 ) (network.Network, error) {
-	return generateDefaultNetwork(log, binaryPath, api.NewAPIClient, NewNodeProcess)
+	return newDefaultNetwork(log, binaryPath, api.NewAPIClient, NewNodeProcess)
 }
 
-// generateDefaultNetwork creates a default network of 5 validator nodes
-// Pre-funded addresses:
-// X chain X-custom18jma8ppw3nhx5r4ap8clazz0dps7rv5u9xde7p
-// privateKey PrivateKey-ewoqjP7PxY4yr3iLTpLisriqt94hdyDFNgchSxGGztUrTXtNN
-// X chain X-custom16045mxr3s2cjycqe2xfluk304xv3ezhkhsvkpr
-// privateKey PrivateKey-2fzYBh3bbWemKxQmMfX6DSuL2BFmDSLQWTvma57xwjQjtf8gFq
-// C chain 0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC
-// privateKey 56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027
-func generateDefaultNetwork(
+func newDefaultNetwork(
 	log logging.Logger,
 	binaryPath string,
 	newAPIClientF api.NewAPIClientF,
 	newNodeProcessF NewNodeProcessF,
 ) (network.Network, error) {
-	for i := 0; i < len(DefaultNetworkConfig.NodeConfigs); i++ {
-		DefaultNetworkConfig.NodeConfigs[i].ImplSpecificConfig = NodeConfig{
+	config := defaultNetworkConfig
+	// Don't overwrite [DefaultNetworkConfig.NodeConfigs]
+	config.NodeConfigs = make([]node.Config, len(defaultNetworkConfig.NodeConfigs))
+	copy(config.NodeConfigs, defaultNetworkConfig.NodeConfigs)
```

Comment:
something is redundant? either the copy func or the element copy inside the for iteration. am i missing something?

Reply:
you're right, removed the redundant line